### PR TITLE
adding doc for azure managed prometheus support in prometheus scaler (#1076

### PR DESCRIPTION
**Pull request by raggupta-ms**
Prometheus scaler now supports Azure managed service for Prometheus. This doc update explains how the prometheus `ScaledObject` can be configured with Azure Auth, query endpoint etc.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes #

### CODE REVIEW

**Comment by raggupta-ms** [comment]
@tomkerkhove @JorTurFer Please take a look at this doc update.

File: content/docs/2.10/scalers/prometheus.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ triggers:
 
 ### Authentication Parameters
 
-Prometheus Scaler supports four types of authentication - bearer authentication, basic authentication, TLS authentication and custom authentication.
+Prometheus Scaler supports five types of authentication - bearer authentication, basic authentication, TLS authentication, custom authentication and Azure Workload/Pod Identity for Azure managed service for Prometheus.
```
Comment (tomkerkhove) [inline content/docs/2.10/scalers/prometheus.md:46]: Given the duplication, I think we can change this to:
```suggestion
Prometheus Scaler supports various types of authentication to help you integrate with Prometheus.
```

File: content/docs/2.10/scalers/prometheus.md [hunk 2]
```diff
@@ -69,7 +69,10 @@ You can use `TriggerAuthentication` CRD to configure the authentication. It is p
 
 > 💡 **NOTE:**It's also possible to set the CA certificate regardless of the selected `authModes` (also without any authentication). This might be useful if you are using an enterprise CA.
 
-### Example
+**Auth for Azure managed service for Prometheus:**
+- See the `Azure managed service for Prometheus` section later on this page.
```
Comment (tomkerkhove) [inline content/docs/2.10/scalers/prometheus.md:73]: We typically provide an overview here to help end-users understand what is required, samples are only a nice to have.

I think the unique fields for this feature is `azureManagedPrometheusResourceURL` and `cloud` if I'm not mistaken?

File: content/docs/2.10/scalers/prometheus.md [hunk 3]
```diff
@@ -351,3 +354,49 @@ spec:
...
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: azure-managed-prometheus-trigger-auth
+spec:
+  podIdentity:
+      provider: azure | azure-workload # use "azure" for pod identity and "azure-workload" for workload identity
+      identityId: <identity-id> # Optional. Default: Identity linked with the label set when installing KEDA.
+
```
Comment (tomkerkhove) [inline content/docs/2.10/scalers/prometheus.md:377]: Nit
```suggestion
```

**Review by tomkerkhove — COMMENTED** [review]
Added some comments but can you also update "Parameter list" + the overview yaml at the top please?

File: content/docs/2.10/scalers/prometheus.md (hunk 1 above)
Comment (raggupta-ms) [reply content/docs/2.10/scalers/prometheus.md]: done

File: content/docs/2.10/scalers/prometheus.md (hunk 2 above)
Comment (raggupta-ms) [reply content/docs/2.10/scalers/prometheus.md]: done

File: content/docs/2.10/scalers/prometheus.md (hunk 3 above)
Comment (raggupta-ms) [reply content/docs/2.10/scalers/prometheus.md]: done

File: content/docs/2.10/scalers/prometheus.md [hunk 4]
```diff
@@ -69,7 +78,15 @@ You can use `TriggerAuthentication` CRD to configure the authentication. It is p
 
 > 💡 **NOTE:**It's also possible to set the CA certificate regardless of the selected `authModes` (also without any authentication). This might be useful if you are using an enterprise CA.
 
-### Example
+### Azure managed service for Prometheus
```
Comment (tomkerkhove) [inline content/docs/2.10/scalers/prometheus.md:81]: Let's align with existing format above
```suggestion
**Azure managed service for Prometheus:**
```

File: content/docs/2.10/scalers/prometheus.md [hunk 5]
```diff
@@ -25,7 +25,14 @@ triggers:
...
     ignoreNullValues: false # Default is `true`, which means ignoring the empty value list from Prometheus. Set to `false` the scaler will return error when Prometheus target is lost
-    unsafeSsl: "false" #  Default is `false`, Used for skipping certificate check when having self signed certs for Prometheus endpoint
+    unsafeSsl: "false" #  Default is `false`, Used for skipping certificate check when having self signed certs for Prometheus endpoint    
+    # Required when using Azure managed service for Prometheus
+    authenticationRef:
+    # Valid when using Azure managed service for Prometheus
+    cloud: Private # Default is `AzurePublicCloud`
+    # Required when cloud = Private
+    azureManagedPrometheusResourceURL: https://prometheus.monitor.azure.airgap/.default
+
```
Comment (tomkerkhove) [inline content/docs/2.10/scalers/prometheus.md:35]: Let's leave this out and cover it in the details
```suggestion
    # Valid when using Azure managed service for Prometheus
    cloud: Private # Default is `AzurePublicCloud`
    # Required when cloud = Private
    azureManagedPrometheusResourceURL: https://prometheus.monitor.azure.airgap/.default
```

**Review by tomkerkhove — APPROVED** [review]

File: content/docs/2.10/scalers/prometheus.md (hunk 4 above)
Comment (raggupta-ms) [reply content/docs/2.10/scalers/prometheus.md]: done

File: content/docs/2.10/scalers/prometheus.md (hunk 5 above)
Comment (raggupta-ms) [reply content/docs/2.10/scalers/prometheus.md]: done

**Comment by raggupta-ms** [comment]
@tomkerkhove what is the process? should I resolve all comments and merge the PR or you would do that?

**Comment by zroubalik** [comment]
> @tomkerkhove what is the process? should I resolve all comments and merge the PR or you would do that?

We will merge all PRs at once.

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**